lib/ClangImporter/ImportType.cpp

```diff
@@ -2227,6 +2227,157 @@ ImportedType ClangImporter::Implementation::importFunctionParamsAndReturnType(
   return {swiftResultTy, importedType.isImplicitlyUnwrapped()};
 }
 
+static ImportTypeKind
+getImportTypeKindForParam(const clang::ParmVarDecl *param) {
+  ImportTypeKind importKind = ImportTypeKind::Parameter;
+  if (param->hasAttr<clang::CFReturnsRetainedAttr>())
+    importKind = ImportTypeKind::CFRetainedOutParameter;
+  else if (param->hasAttr<clang::CFReturnsNotRetainedAttr>())
+    importKind = ImportTypeKind::CFUnretainedOutParameter;
+
+  return importKind;
+}
+
+Optional<swift::Type> ClangImporter::Implementation::importParameterType(
+    const clang::ParmVarDecl *param, OptionalTypeKind optionalityOfParam,
+    ImportTypeKind importKind, bool allowNSUIntegerAsInt,
+    bool isNSDictionarySubscriptGetter, bool paramIsError,
+    bool paramIsCompletionHandler,
+    Optional<unsigned> completionHandlerErrorParamIndex,
+    ArrayRef<GenericTypeParamDecl *> genericParams,
+    llvm::function_ref<void(Diagnostic &&)> addImportDiagnosticFn,
+    bool &isInOut, bool &isParamTypeImplicitlyUnwrapped) {
+  auto paramTy = param->getType();
```

Comment:
One observation, the following here:
```
  auto paramTy = param->getType();
  auto paramAddDiag = addImportDiagnosticFn;
```

could make the diff for this function a lot more readable.

Reply:
I can rename the parameter to `paramAddDiag`. I thought that keeping `addImportDiagnosticFn` like in other functions was a better idea, but I also was not convinced by the diff being a little bit more broken because of it.